### PR TITLE
Addresses #828 Sync locked details onto subjects

### LINKED ISSUE #828: Sync locked details onto subjects
You can lock conversations on GitHub and provide a reason why: https://help.github.com/articles/locking-conversations/

We should add a `locked` field to the `subjects` table and update that field when syncing subjects.

Then we can add a search prefix for filtering `locked:true` or `locked:false`

### PR DESCRIPTION
**Pull request by prashuchaudhary**
Fixes #828

### CODE REVIEW

**Comment by prashuchaudhary** [comment]
@andrew please review the changes

File: app/models/search.rb [hunk 1]
```diff
@@ -24,6 +24,7 @@ def results
     res = res.bot_author unless bot_author.nil?
     res = res.unlabelled unless unlabelled.nil?
     res = res.is_private(is_private) unless is_private.nil?
+    res = res.locked unless is_locked.nil? || !is_locked
```
Comment (andrew) [inline app/models/search.rb]: Does this work for `locked:false`?

Comment (prashuchaudhary) [reply app/models/search.rb]: yes. it works for that too

**Comment by prashuchaudhary** [comment]
@andrew will add test cases

**Comment by prashuchaudhary** [comment]
Added unit test case for Locked flag.

**Comment by andrew** [comment]
Thanks again @prashuchaudhary 👏

**Pull request merged**